centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -718,40 +719,57 @@ object Operations extends StrictLogging {
-            "value" -> value) ++ old.map(o => "old" -> o))
-          case diffson.jsonpatch.Test(path, value) => JsObject(Map[String, JsValue](
-            "op" -> JsString("test"),
-            "path" -> JsString(path.toString),
-            "value" -> value))
-        }
+      // Sometimes new metadata is allowed to be added between the diffs. Account for that here:
+      def isAllowableAddition(pointer: jsonpointer.Pointer): Boolean = pointer.parts.toList match {
+        case Left(x) :: _ if allowableAddedOneWordFields.contains(x) => true
+        case Left("calls") :: Left(_) :: Right(_) :: Left(x) :: _ if allowableAddedOneWordFields.contains(x) => true
+        case _ => false
+      }
 
-        override def write(obj: JsonPatch[JsValue]): JsValue = {
-          JsArray(obj.ops.toVector.map(processOperation))
-        }
+      val filteredDifferences = diff(expected: JsValue, actual: JsValue).ops.toVector filter {
+        case Add(path, _) if isAllowableAddition(path) => false // Exclude valid additions
+        case _ => true // Include anything else
       }
 
-      val jsonDiff = diff(expected: JsValue, actual: JsValue).toJson.prettyPrint
-      IO.raiseError(CentaurTestException(s"Error during $testType metadata comparison. Diff: $jsonDiff Expected: $expected Actual: $actual", workflow, submittedWorkflow))
+      if (filteredDifferences.isEmpty) {
+        IO.unit
+      } else {
+        val writer: JsonWriter[Vector[Operation[JsValue]]] = new JsonWriter[Vector[Operation[JsValue]]] {
+          def processOperation(op: Operation[JsValue]): JsValue = op match {
+            case Add(path, value) => JsObject(Map[String, JsValue](
+              "description" -> JsString("Unexpected value found"),
+              "path" -> JsString(path.toString),
+              "value" -> value))
+            case Copy(from, path) => JsObject(Map[String, JsValue](
+              "description" -> JsString("Value(s) unexpectedly copied"),
+              "expected_at" -> JsString(from.toString),
+              "also_at" -> JsString(path.toString)))
+            case Move(from, path) => JsObject(Map[String, JsValue](
+              "description" -> JsString("Value(s) unexpectedly moved"),
+              "expected_location" -> JsString(from.toString),
+              "actual_location" -> JsString(path.toString)))
+            case Remove(path, old) => JsObject(Map[String, JsValue](
+              "description" -> JsString("Value missing"),
+              "expected_location" -> JsString(path.toString)) ++
+              old.map(o => "expected_value" -> o))
+            case Replace(path, value, old) => JsObject(Map[String, JsValue](
+              "description" -> JsString("Incorrect value found"),
+              "path" -> JsString(path.toString),
+              "found_value" -> value) ++ old.map(o => "expected_value" -> o))
+            case diffson.jsonpatch.Test(path, value) => JsObject(Map[String, JsValue](
```

Comment:
Not surprised at the non-coverage as I don't understand how we would see a [Test](https://github.com/gnieh/diffson/blob/7611e514884c7ba7ea88e5ea98c78246bfbde7cc/core/src/main/scala/diffson/jsonpatch/JsonPatch.scala#L173) resulting from a diff operation?

Reply:
Yeah, I think this is more for exhaustively satisfying the match than something I'd actually expect us to see